### PR TITLE
Feature/ceres governance upgrade

### PR DESCRIPTION
**Pull request by JovanCBS**
Upgrade Ceres governance platform.

### CODE REVIEW

File: pallets/ceres-governance-platform/src/weights.rs
Comment (wer1st) [inline pallets/ceres-governance-platform/src/weights.rs:1]: Did you measure it following our process and used special reference machine?

File: pallets/ceres-governance-platform/src/benchmarking.rs [hunk 1]
```diff
@@ -30,101 +24,142 @@ fn assert_last_event<T: Config>(generic_event: <T as Config>::RuntimeEvent) {
...
+        let nonce = frame_system::Pallet::<T>::account_nonce(&caller);
+        let encoded: [u8; 32] = (&caller, nonce).using_encoded(blake2_256);
+        let poll_id = H256::from(encoded);
+        let mut options = BoundedVec::default();
+        options.try_push("Yes".try_into().unwrap()).unwrap();
+        options.try_push("No".try_into().unwrap()).unwrap();
+
         frame_system::Pallet::<T>::inc_providers(&caller);
+
+        let owner: T::AccountId = assets::AssetOwners::<T>::get::<T::AssetId>(asset_id.into()).unwrap();
```
Comment (wer1st) [inline pallets/ceres-governance-platform/src/benchmarking.rs:92]: Benchmark should test the worst case. Please set the max possible count for options, max length strings for title and description.
Options also should have names are filled up to the max length.

File: pallets/ceres-governance-platform/src/migrations.rs
[the remark this answers is not in the document]
Comment (JovanCBS) [reply pallets/ceres-governance-platform/src/migrations.rs:1]: @vovac12 did it in test environment with mainnet snapshot.

File: pallets/ceres-governance-platform/src/tests.rs [hunk 2]
```diff
@@ -1,28 +1,48 @@
+use crate::migrations::{OldPollInfo, OldVotingInfo};
 use crate::mock::*;
 use crate::{pallet, Error};
-use common::{balance, generate_storage_instance, AssetInfoProvider, CERES_ASSET_ID};
-use frame_support::pallet_prelude::StorageMap;
+use codec::Decode;
+use codec::Encode;
```
[the remark this answers is not in the document]
Comment (JovanCBS) [reply pallets/ceres-governance-platform/src/tests.rs:5]: Done.

File: pallets/ceres-governance-platform/src/weights.rs
Comment (JovanCBS) [reply pallets/ceres-governance-platform/src/weights.rs:1]: No.

File: pallets/ceres-governance-platform/src/benchmarking.rs (hunk 1 above)
Comment (JovanCBS) [reply pallets/ceres-governance-platform/src/benchmarking.rs]: Done.

**Review by wer1st — APPROVED** [review]

File: pallets/ceres-governance-platform/src/lib.rs [hunk 3]
```diff
@@ -12,38 +13,56 @@ mod mock;
...
 pub use weights::WeightInfo;
 
 #[derive(Encode, Decode, Default, PartialEq, Eq, scale_info::TypeInfo)]
 #[cfg_attr(feature = "std", derive(Debug))]
-pub struct VotingInfo {
+#[scale_info(skip_type_params(StringLimit))]
+pub struct VotingInfo<StringLimit: sp_core::Get<u32>> {
     /// Voting option
-    voting_option: u32,
+    voting_option: BoundedString<StringLimit>,
```
Comment (vovac12) [inline pallets/ceres-governance-platform/src/lib.rs:25]: Why not use the option id? You don't really need to save the full option name for each vote

File: pallets/ceres-governance-platform/src/migrations.rs [hunk 4]
```diff
@@ -1,46 +1,191 @@
...
+    AccountIdOf<T>,
+    OldVotingInfo,
+    ValueQuery,
+>;
+
+pub fn migrate<T: Config>() -> Result<(), &'static str> {
+    let poll_asset: AssetIdOf<T> = CERES_ASSET_ID.into();
+    let user = AuthorityAccount::<T>::get();
+    let bytes = hex!("c4e7d5a63d8e887932bb6dc505dd204005c3ecfb6de5f1f0d3ac0a308b2b2915");
+    let first_poll_creator = AccountIdOf::<T>::decode(&mut &bytes[..]).unwrap();
```
[the remark this answers is not in the document]
Comment (JovanCBS) [reply pallets/ceres-governance-platform/src/migrations.rs:47]: Done.

File: pallets/ceres-governance-platform/src/lib.rs (hunk 3 above)
Comment (JovanCBS) [reply pallets/ceres-governance-platform/src/lib.rs:25]: Done with Index as voting_option.

**Review by vovac12 — APPROVED** [review]

**Pull request merged**